From 01ac25a8e00178fc49ea949ef646c3e9807a8288 Mon Sep 17 00:00:00 2001
From: Ludovic Henry <git@ludovic.dev>
Date: Tue, 13 Oct 2026 18:44:31 +0000
Subject: [PATCH 2/2] Refresh config.guess/config.sub for the autotools
 3rdparty builds

The FFTW 3.3.2, libyaml 0.1.5, libsamplerate 0.1.9 and LAME 3.100
release tarballs ship config.guess/config.sub copies dated 2009-2015,
which predate the riscv64 triplet, so their configure scripts stop with
"cannot guess build type; you must specify one". Passing --build does
not help because the same stale config.sub rejects the triplet. Copy the
build image's automake copies over them before running configure.

Upstream-Status: Inappropriate [relies on the build image shipping a current automake under /usr/share; only needed to build these old tarballs on riscv64]

Signed-off-by: Ludovic Henry <git@ludovic.dev>
---
 packaging/debian_3rdparty/build_fftw3.sh         | 1 +
 packaging/debian_3rdparty/build_lame.sh          | 1 +
 packaging/debian_3rdparty/build_libsamplerate.sh | 1 +
 packaging/debian_3rdparty/build_yaml.sh          | 1 +
 4 files changed, 4 insertions(+)

diff --git a/packaging/debian_3rdparty/build_fftw3.sh b/packaging/debian_3rdparty/build_fftw3.sh
index 846101e1..8f46f5ac 100755
--- a/packaging/debian_3rdparty/build_fftw3.sh
+++ b/packaging/debian_3rdparty/build_fftw3.sh
@@ -11,6 +11,7 @@ echo "Building fftw $FFTW_VERSION"
 curl -SLO http://www.fftw.org/$FFTW_VERSION.tar.gz
 tar -xf $FFTW_VERSION.tar.gz
 cd $FFTW_VERSION
+cp /usr/share/automake-*/config.guess /usr/share/automake-*/config.sub ./
 
 CPPFLAGS=-fPIC ./configure \
     --prefix=$PREFIX \
diff --git a/packaging/debian_3rdparty/build_lame.sh b/packaging/debian_3rdparty/build_lame.sh
index c363dc9d..8915274b 100755
--- a/packaging/debian_3rdparty/build_lame.sh
+++ b/packaging/debian_3rdparty/build_lame.sh
@@ -12,6 +12,7 @@ echo "Building lame $LAME_VERSION"
 curl -SL -o lame-$LAME_VERSION.tar.gz "http://downloads.sourceforge.net/project/lame/lame/$LAME_VERSION/lame-$LAME_VERSION.tar.gz?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Flame%2F&ts=1476009914&use_mirror=ufpr"
 tar -xf  lame-$LAME_VERSION.tar.gz
 cd lame-$LAME_VERSION
+cp /usr/share/automake-*/config.guess /usr/share/automake-*/config.sub ./
 CPPFLAGS=-fPIC ./configure --prefix=$PREFIX \
     $SHARED_OR_STATIC
 make
diff --git a/packaging/debian_3rdparty/build_libsamplerate.sh b/packaging/debian_3rdparty/build_libsamplerate.sh
index ed20edcd..f3e38ad4 100755
--- a/packaging/debian_3rdparty/build_libsamplerate.sh
+++ b/packaging/debian_3rdparty/build_libsamplerate.sh
@@ -11,6 +11,7 @@ echo "Building libsamplerate $LIBSAMPLERATE_VERSION"
 curl -SLO https://github.com/libsndfile/libsamplerate/releases/download/0.1.9/$LIBSAMPLERATE_VERSION.tar.gz
 tar -xf $LIBSAMPLERATE_VERSION.tar.gz
 cd $LIBSAMPLERATE_VERSION
+cp /usr/share/automake-*/config.guess /usr/share/automake-*/config.sub ./Cfg/
 
 CPPFLAGS=-fPIC ./configure \
     --prefix=$PREFIX \
diff --git a/packaging/debian_3rdparty/build_yaml.sh b/packaging/debian_3rdparty/build_yaml.sh
index 9a87e049..e4cc9a83 100755
--- a/packaging/debian_3rdparty/build_yaml.sh
+++ b/packaging/debian_3rdparty/build_yaml.sh
@@ -11,6 +11,7 @@ echo "Building libyaml $LIBYAML_VERSION"
 curl -SLO http://pyyaml.org/download/libyaml/$LIBYAML_VERSION.tar.gz
 tar -xf $LIBYAML_VERSION.tar.gz
 cd $LIBYAML_VERSION
+cp /usr/share/automake-*/config.guess /usr/share/automake-*/config.sub ./config/
 
 # fails to compile shared library, building only static
 CFLAGS="-DYAML_DECLARE_STATIC -fPIC" ./configure \
-- 
2.43.0

